src/serializers/ob_type.rs

```diff
@@ -82,7 +82,12 @@ impl ObTypeLookup {
             timedelta: PyDelta::new(py, 0, 0, 0, false).unwrap().get_type_ptr() as usize,
             url: PyUrl::new(lib_url.clone()).into_py(py).as_ref(py).get_type_ptr() as usize,
             multi_host_url: PyMultiHostUrl::new(lib_url, None).into_py(py).as_ref(py).get_type_ptr() as usize,
-            enum_object: py.import("enum").unwrap().getattr("Enum").unwrap().to_object(py),
+            enum_object: py
+                .import(intern!(py, "enum"))
+                .unwrap()
+                .getattr(intern!(py, "EnumMeta"))
```

Comment:
This change looks unrelated and a possible bug, but I'm surprised that nothing broke?

```suggestion
                .getattr(intern!(py, "Enum"))
```

Reply:
@davidhewitt I can revert but it seems like a bug to me. Here is repro:
```py
>>> from enum import Enum, EnumMeta
>>> class PlainEnum(Enum):
...     ONE = 'one'
... 
>>> [type(type(PlainEnum.ONE)) is t for t in (Enum, EnumMeta)]
[False, True]
>>> type(type(PlainEnum.ONE))
<class 'enum.EnumMeta'>
>>> class StrEnum(str, Enum):
...     ONE = 'one'
... 
>>> [type(type(StrEnum.ONE)) is t for t in (Enum, EnumMeta)]
[False, True]
>>> type(type(StrEnum.ONE))
<class 'enum.EnumMeta'>
```

When I was implementing this, I could never get:
https://github.com/pydantic/pydantic-core/blob/23d106551d14214f24a0bcd577b74a672ccfd0c0/src/serializers/ob_type.rs#L259-L267
to return `True` when importing just `Enum`

Reply:
Right so that function is definitely bugged, it looks like in other places we do expect to check `isinstance(x, Enum)` so I think rather than changing `enum_object` here we would need `enum_meta_object` as a separate type.

Alternatively how about:

```rust
fn is_enum(&self, op_value: Option<&PyAny>, py_type: &PyType) -> bool { 
     // only test on the type itself, not base types 
     if op_value.is_some() { 
         let enum_meta = self.enum_object.get_type();
         let meta_type = py_type.get_type(); 
         meta_type.is(enum_meta) 
     } else { 
         false 
     } 
 } 
```